app/controllers/IncomeSourceSummaryController.scala

```diff
@@ -68,40 +70,44 @@ class IncomeSourceSummaryController @Inject() (
-            taxCodeIncomes,
-            payments,
-            estimatedPayCompletion,
-            cacheUpdatedIncomeAmount
-          ) =>
-        val incomeDetailsViewModel = IncomeSourceSummaryViewModel.applyNew(
-          empId = empId,
-          displayName = request.fullName,
-          taxCodeIncomes.fold(_ => None, _.find(_.employmentId.fold(false)(_ == employment.sequenceNumber))),
-          employment = employment,
-          payments = payments.toOption.flatMap(_.find(_.sequenceNumber == employment.sequenceNumber)),
-          estimatedPayJourneyCompleted = estimatedPayCompletion,
-          rtiAvailable = isRTIAvailable(payments),
-          cacheUpdatedIncomeAmount = cacheUpdatedIncomeAmount
-        )
+    empIdCheck.checkValidId(empId).flatMap {
+      case Some(result) => Future.successful(result)
+      case _            =>
+        (
+          employmentService.employmentOnly(nino, empId, TaxYear()),
+          taxAccountService.taxCodeIncomes(nino, TaxYear()),
+          rtiService.getPaymentsForYear(nino, TaxYear()).value,
+          Future.successful(hasJourneyCompleted),
+          cacheUpdatedIncomeAmountFuture
+        ).mapN {
+          case (
+                Some(employment),
+                taxCodeIncomes,
+                payments,
+                estimatedPayCompletion,
+                cacheUpdatedIncomeAmount
+              ) =>
+            val incomeDetailsViewModel = IncomeSourceSummaryViewModel.applyNew(
+              empId = empId,
+              displayName = request.fullName,
+              taxCodeIncomes.fold(_ => None, _.find(_.employmentId.fold(false)(_ == employment.sequenceNumber))),
+              employment = employment,
+              payments = payments.toOption.flatMap(_.find(_.sequenceNumber == employment.sequenceNumber)),
+              estimatedPayJourneyCompleted = estimatedPayCompletion,
+              rtiAvailable = isRTIAvailable(payments),
+              cacheUpdatedIncomeAmount = cacheUpdatedIncomeAmount
+            )
 
-        if (!incomeDetailsViewModel.isUpdateInProgress) {
-          val updatedUserAnswers = request.userAnswers.remove(UpdateIncomeConfirmedNewAmountPage(empId))
-          journeyCacheRepository.set(updatedUserAnswers)
-        }
+            if (!incomeDetailsViewModel.isUpdateInProgress) {
+              val updatedUserAnswers = request.userAnswers.remove(UpdateIncomeConfirmedNewAmountPage(empId))
+              journeyCacheRepository.set(updatedUserAnswers)
+            }
 
-        Ok(incomeSourceSummary(incomeDetailsViewModel))
-      case _ => errorPagesHandler.internalServerError("Error while fetching income summary details")
-    } recover { case NonFatal(e) =>
-      errorPagesHandler.internalServerError("IncomeSourceSummaryController exception", Some(e))
+            Ok(incomeSourceSummary(incomeDetailsViewModel))
+          case _ => errorPagesHandler.internalServerError("Error while fetching income summary details")
```

Comment:
This should either be Error while fetching employment or better should not it take user to page not found types because employment does not exist